instantmusic-0.1/bin/instantmusic

```diff
@@ -114,7 +114,7 @@ def query_and_download(search, has_prompts=True, is_quiet=False):
     # We only ask the user which one they want if prompts are on, of course
     if has_prompts and not is_quiet:
         choice = ''
-        while choice.strip() == '':
+        while not(choice.isdigit()) or not(0 <= int(choice) < len(available)):
```

Comment:
I actually think the current solution would be nicer. Reprinting the "Pick one: " one prompt should be enough as a hint to the user that something went wrong. 
But we could do something like:

```
choice = raw_input('Pick one: ')
while not(choice.isdigit()) or not(0 <= int(choice) < len(available)):
    print("Oups, that was wrong. Try again!")
    choice = raw_input('Pick one: ')
```